src/utils/hibp.ts

```diff
@@ -0,0 +1,443 @@
+// We use 'Name' in Firefox (via Remote Settings), so we have to maintain our own mapping of re-named breaches.
+const RENAMED_BREACHES = ["covve"];
+const RENAMED_BREACHES_MAP = {
+  covve: "db8151dd",
+};
+
+// TODO: Add unit test when changing this code:
+/* c8 ignore start */
+function _addStandardOptions(options = {}) {
+  const hibpOptions = {
+    headers: {
+      "User-Agent": HIBP_USER_AGENT,
+    },
+    ...options,
+  };
+  return hibpOptions;
+}
+/* c8 ignore stop */
+
+// TODO: Add unit test when changing this code:
+/* c8 ignore start */
+async function _throttledFetch(
+  url: string,
+  reqOptions?: RequestInit,
+  tryCount = 1,
+) {
+  try {
+    const response = await fetch(url, reqOptions);
+    if (response.ok) return (await response.json()) as unknown;
+
+    switch (response.status) {
+      case 404:
+        // 404 can mean "no results", return undefined response
+        console.info("_throttledFetch", {
+          err: "Error 404, not going to retry. TryCount: " + tryCount,
+        });
+        return undefined;
+      case 429:
+        console.info("_throttledFetch", {
+          err: "Error 429, tryCount: " + tryCount,
+        });
+        // @ts-ignore TODO: Explicitly parse into a number
+        if (tryCount >= HIBP_THROTTLE_MAX_TRIES) {
+          throw new InternalServerError(getMessage("error-hibp-throttled"));
+        } else {
+          tryCount++;
+          await new Promise((resolve) =>
+            // @ts-ignore HIBP_THROTTLE_DELAY should be defined
+            setTimeout(resolve, HIBP_THROTTLE_DELAY * tryCount),
+          );
+          return await _throttledFetch(url, reqOptions, tryCount);
+        }
+      default:
+        console.error(await response.text());
+        throw new InternalServerError(`bad response: ${response.status}`);
+    }
+  } catch (err) {
+    console.error("_throttledFetch", { err });
+    throw new InternalServerError(getMessage("error-hibp-connect"));
+  }
+}
+/* c8 ignore stop */
+
+// TODO: Add unit test when changing this code:
+/* c8 ignore start */
+async function hibpApiFetch(path: string, options = {}) {
+  const url = `${HIBP_API_ROOT}${path}`;
+  const reqOptions = _addStandardOptions(options);
+  try {
+    return await _throttledFetch(url, reqOptions);
+  } catch (ex) {
+    console.error(ex);
+  }
+}
+/* c8 ignore stop */
+
+export type HibpGetBreachesResponse = Array<{
+  Name: string;
+  Title: string;
+  Domain: string;
+  AddedDate: ISO8601DateString;
+  BreachDate: ISO8601DateString;
+  DataClasses: Array<keyof HibpBreachDataTypes>;
+  Description: string;
+  LogoPath: string;
+  IsFabricated: boolean;
+  IsMalware: boolean;
+  IsRetired: boolean;
+  IsSensitive: boolean;
+  IsSpamList: boolean;
+  IsVerified: boolean;
+  ModifiedDate: ISO8601DateString;
+  PwnCount: number;
+}>;
+
+export async function fetchHibpBreaches(): Promise<HibpGetBreachesResponse> {
+  return (await hibpApiFetch("/breaches")) as HibpGetBreachesResponse;
+}
```

Comment:
Looks like the changes were too hard for Git to reconcile across the `.js` and `.ts` file, so let me try to point them out.

Here I renamed `req` to `hibpApiFetch` and made it internal to this module. As called out before, I added a `fetchHibpBreaches` wrapper that specifically calls out to HIBP's `/breaches` API and types the return value.